### PR TITLE
update circleci to latest orb, add yamllint and mdl

### PR DESCRIPTION
**Pull request by JohnRoesler**
### Description

update circleci to latest orb, add yamllint and mdl

### Issues Resolved

[List any existing issues this PR resolves]

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by Xorima** [comment]
Remember to do the terraform pull request for this :)

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,53 +1,65 @@
...
-      - kitchen/lint:
-          name: lint
+      - kitchen/delivery:
+          name: delivery
       - kitchen/dokken-single:
           name: server-debian-8
-          suite: server
-          platform: debian-8
-          requires: [danger, lint]
+          suite: server-debian-8
```
Comment (Xorima) [inline .circleci/config.yml]: Why specific, you could just do server and have it cover them all, I guess it's how granular you want it?

Comment (JohnRoesler) [reply .circleci/config.yml]: Yeah - I was just updating based on what was there - don't know a ton about this repo so I didn't want to mess with the test setup if it was granular for a reason

**Comment by JohnRoesler** [comment]
TF PR https://github.com/sous-chefs/terraform-github-org/pull/179

**Pull request merged**